src/openms_gui/source/VISUAL/TVSpectraViewController.cpp

```diff
@@ -236,13 +237,14 @@ namespace OpenMS
       // first get raw data (the full experiment with all chromatograms), we
       // only need to grab the ones with the desired indices
       ExperimentSharedPtrType exp_sptr = layer.getChromatogramData();
-      auto ondisc_sptr = layer.getOnDiscPeakData();
 
-      widget_1d->canvas()->removeLayers();
+      String fname = layer.filename;
+      auto annotation = layer.getChromatogramAnnotation();
+      auto ondisc_sptr = layer.getOnDiscPeakData();
+      widget_1d->canvas()->removeLayers(); // this actually deletes layer, make sure its not used any more
```

Comment:
random segfaults result when `layer` is used after calling `widget_1d->canvas()->removeLayers();`

Reply:
nice fix